tests/system/test_context_lifecycle_records.py

```
@@ -24,7 +24,84 @@ def test_context_window_trim_preserves_llm_request_lifecycle_records() -> None:
+
+
+def test_image_trimming_counts_image_parts_not_messages() -> None:
+    cwm = ContextWindowManager(token_counter=lambda content: len(str(content)))
+    cwm.max_context_images = 2
+
+    session = Session()
+    old_message = Message(
+        role="user",
+        content=[
+            {"type": "text", "text": "old"},
+            {"type": "image_url", "image_path": "/tmp/one.png"},
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/one.png"

(S108)